pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -1685,12 +1674,15 @@ func appNetworkDoInactivateUnderlayNetwork(
 		ntpServers := types.GetNTPServers(*ctx.deviceNetworkStatus,
 			netstatus.CurrentUplinkIntf)
 		createDnsmasqConfiglet(ctx, bridgeName,
-			ulStatus.BridgeIPAddr, netconfig, hostsDirpath,
+			ulStatus.BridgeIPAddr, netstatus, hostsDirpath,
 			newIpsets, netstatus.CurrentUplinkIntf,
 			dnsServers, ntpServers)
 		startDnsmasq(bridgeName)
 	}
 	netstatus.RemoveVif(log, ulStatus.Vif)
+	if maybeNetworkInstanceDelete(ctx, netstatus) {
```

Comment:
Do we need a flag in the NetworkInstanceStatus to mark it as TO_BE_DELETED.
Or Is the intention to go ahead and delete netowrk instances that do not have any vifs attached irrespective of the network instance still being present in the device configuration?

Reply:
Good point; and that might explain why some tests are failing.

Reply:
Fixed inside the maybe function by looking for Config.